### PR TITLE
Separate trace and span status enum

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11754?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11754/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11754
```

</p>
</details>

### Related Issues/PRs

Follow-up of https://github.com/mlflow/mlflow/pull/11734#discussion_r1569740014

### What changes are proposed in this pull request?

Currently, both trace and span shares the same set of status code defined in the `TraceStatus` enum. However, the new `IN_PROGRESS` status was added to trace status, which is not a valid (otel-compatible) status for the span. Hence, this PR creates separate set of status code for spans.

Note that users still only interact with a single enum, which is `SpanStatusCode`.  The other enum `TraceStatus` should always be managed by us and won't be surfaced in users code.

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by B-Step62** [comment]
Not sure why LangChain model tests fail but they seem to be unrelated

```
=========================== short test summary info ============================
FAILED | MEM 2.0/15.6 GB | DISK 60.2/72.5 GB tests/langchain/test_langchain_model_export.py::test_langchain_log_huggingface_hub_model_metadata - 
mlflow.exceptions.MlflowException: Unable to determine if the value supplied by the argument torch_dtype is valid since torch is not installed.
```

**Review by harupy — APPROVED** [review]
LGTM!

**Review by BenWilson2 — APPROVED** [review]

File: mlflow/entities/trace_status.py [hunk 1]
```diff
@@ -17,3 +19,19 @@ def to_proto(self):
     @staticmethod
     def from_proto(proto_status):
         return ProtoTraceStatus.Name(proto_status)
+
+    def to_otel_status(self) -> trace_api.Status:
+        return trace_api.Status(_MLFLOW_STATUS_CODE_TO_OTEL[self])
```
Comment (daniellok-db) [inline mlflow/entities/trace_status.py:24]: nit/question: should we map `IN_PROGRESS` to `UNSPECIFIED`? what happens at the moment if we try to call this function when the value is `IN_PROGRESS`?

Comment (B-Step62) [reply mlflow/entities/trace_status.py:24]: Oh great catch! Actually we don't need `to_otel_status()`, we only do the opposite mapping. Will remove this function.

**Review by daniellok-db — APPROVED** [review]
lgtm

**Comment by B-Step62** [comment]
It seems we need to cherry-pick https://github.com/mlflow/mlflow/commit/6e41ed57946e24f485c4f82b141bc9b4bbc72574 to resolve LangChain test failures. Merging this PR neverthess and will rebase the tracing branch.

**Pull request merged**